### DISCUSSION TITLE
How to let renovate to update the gemspec by bundler

### DISCUSSION THREAD

**Discussion by tata9001** [renovatebot/renovate#10585] [Request Help]
Currently, we have bunch of repo which use the gemspec to manage the dependencies.

After few times test, seems renovate cannot support the gemspec pattern which the dependabot support.

Cannot understand what's meaning of the [comment](https://github.com/renovatebot/renovate/issues/2983#issuecomment-626279510),  renovate can support the gemspec? How?

**Comment by viceice** [comment]
renovate currently only supported updating `Gemfile` file.

Please provide a minimal reproduction so we can have a look. Maybe it's easy to implement.
https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md#about-minimal-reproductions

**Reply by viceice** [reply]
Maybe we can suggest a regex manager as workaround until we support `gemspec`

**Reply by viceice** [reply]
Are you referring to https://guides.rubygems.org/specification-reference/ ?

**Reply by HonkingGoose** [reply]
I think @rarkins closed a issue which covers `gemspec` stuff as `wontfix`: https://github.com/renovatebot/renovate/issues/2983#issuecomment-626279510

We don't currently have an open issue for `gemspec`, and I don't know if we want one. What do you think @rarkins?

**Reply by rarkins** [reply]
I would definitely like to parse `gemspec` files, but only using JS-based parsing and therefore they may need to confirm to "conventions" rather than being completely free of convention. Let's have a feature request issue to discuss it if we don't already

**Reply by tata9001** [reply]
@viceice Yep, you are right.
The part of the gemspec which mange the dependencies is [here](https://guides.rubygems.org/specification-reference/#add_development_dependency): 
- `add_development_dependency` 
- `add_runtime_dependency`(alias: `add_dependency `) 

Here are some OSS repo.
- https://github.com/mdub/config_hound/blob/master/config_hound.gemspec
- https://github.com/realestate-com-au/stackup/blob/master/stackup.gemspec

BTW, you said `a regex manager`, how can write one?
Have you got any demo?

Thanks

**Reply by viceice** [reply]
Sure, in our official docs

https://docs.renovatebot.com/modules/manager/regex/

**Reply by HonkingGoose** [reply]
I've created issue #10616 to discuss the feature request to parse `gemspec` files.

**Comment by mpkorstanje** [comment]
An example of a Regex Manager that can be used to parse Gemspec files.


```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
  ],
  "regexManagers": [
    {
      "fileMatch": [
        "^*.gemspec"
      ],
      "matchStrings": [
        ".*\\.add_(:?(?<depType>.*?)_)?dependency\\s*(['\"])(?<depName>[^'\"]+)(['\"])(\\s*,\\s*(?<currentValue>(['\"])[^'\"]+['\"](\\s*,\\s*['\"][^'\"]+['\"])?))?"
      ],
      "datasourceTemplate": "rubygems",
      "versioningTemplate": "ruby"
    }
  ]
}
```

**Reply by narwold** [reply]
The main issue with this is that it doesn't run `bundle install` afterwards, so that's a manual process for us.
